packages/api-rest/__tests__/apis/common/internalPost.test.ts

```diff
@@ -91,6 +91,27 @@ describe('internal post', () => {
 			expect.objectContaining({ region: 'us-west-2', service: 'lambda' }),
 		);
 	});
+
+	it('should call authenticatedHandler for appsync-api service', async () => {
```

Comment:
adding a missing unit test scenario to get better coverage

Reply:
Can be better described as:
```suggestion
	it('should call authenticatedHandler for appsync-api service with default signing name', async () => {
```

We could add another test case where `signingServiceInfo.service` is specified.

Reply:
Can you add a mock to verify the correct new `iamAuthApplicable` callback is invoked?

Reply:
So the existing tests already cover based on inputs whether authenticated or unthenticatevd handler to be called which implicitly makes sure the `isIamAuthApplicableForGraphql`  is called. I dont see why we need to add a mock to verify the correct callback 🤔

Reply:
Check in offline with Allan, agreed this should suffice.